main.js

```diff
@@ -0,0 +1,165 @@
+// 1
+
+function getDeltaFunction(numbersObject) {
+    let a = numbersObject.a;
+    let b = numbersObject.b;
+    let c = numbersObject.c;
+
+    function getDelta(newNumbersObject = {}) {
+        if (newNumbersObject.a !== undefined) {
+            a = newNumbersObject.a;
+        }
+        if (newNumbersObject.b !== undefined) {
+            b = newNumbersObject.b;
+        }
+        if (newNumbersObject.c !== undefined) {
+            c = newNumbersObject.c;
+        }
+        return (b * b) - 4 * a * c;
+    }
+    return getDelta;
+}
+
+const getDelta = getDeltaFunction({ a: 1, b: 2, c: 3 });
+console.log(getDelta()); // -8
+console.log(getDelta({ b: 12 })); // 132
+console.log(getDelta({ a: 4, b: 10 })); // 52
+console.log(getDelta({ a: 4, b: 0 })); // -48
+
+// The nested function returns NaN if any of the arguments are missing
+console.log(getDeltaFunction({ a: 1, c: 3 })()); // NaN
+console.log(getDeltaFunction({ a: 1, c: 3 })({ b: 15 })); // 213
+
+// 2
+
+function getPositiveSum(numbersArray) {
+    let positiveSum = 0
```

Comment:
There is a missing semicolon here. It's crucial to avoid making the same mistakes over and over.